client/me/notification-settings/settings-form/stream-selector.jsx

```diff
@@ -1,46 +1,37 @@
-import { map } from 'lodash';
 import PropTypes from 'prop-types';
-import { PureComponent } from 'react';
-import { connect } from 'react-redux';
 import FormSelect from 'calypso/components/forms/form-select';
-import getUserDevices from 'calypso/state/selectors/get-user-devices';
+import { useUserDevicesQuery } from 'calypso/data/user-devices/use-user-devices-query';
 import { getLabelForStream } from './locales';
 
-class NotificationSettingsFormStreamSelector extends PureComponent {
-	static propTypes = {
-		devices: PropTypes.array,
-		selectedStream: PropTypes.oneOfType( [ PropTypes.string, PropTypes.number ] ).isRequired,
-		settings: PropTypes.object,
-		onChange: PropTypes.func.isRequired,
-	};
+function NotificationSettingsFormStreamSelector( { selectedStream, onChange } ) {
+	const { data: devices = [] } = useUserDevicesQuery();
 
-	onChange = ( event ) => this.props.onChange( event.target.value );
-
-	render() {
-		const options = [ 'timeline', 'email' ]
-			.map( ( stream ) => (
-				<option key={ stream } value={ stream }>
-					{ getLabelForStream( stream ) }
+	const options = [ 'timeline', 'email' ]
+		.map( ( stream ) => (
+			<option key={ stream } value={ stream }>
+				{ getLabelForStream( stream ) }
+			</option>
+		) )
+		.concat(
+			devices.map( ( device ) => (
+				<option key={ device.id } value={ device.id }>
```

Comment:
I'd suggest prefixing the `key` with something:
```suggestion
				<option key={ `device-notifications-${ device.id }` } value={ device.id }>
```